esda/moran.py

```diff
@@ -1626,6 +1628,34 @@ def vec_permutations_all(max_card: int, n:int, k_replications: int):
         result[k] = numpy.random.choice(n-1, size=max_card, replace=False)
     return result
 
+@njit(fastmath=True)
+def _do_one_observation(i: int,
+                        z: numpy.ndarray,
+                        permuted_ids: numpy.ndarray,
+                        weights: numpy.ndarray,
+                        start: numpy.ndarray,
+                        cardinalities: numpy.ndarray,
+                        mask: numpy.ndarray,
```

Comment:
The jobs can't share the mask, I think.

Reply:
I.e. when you parallelize this, the mask will get modified in each job. I think that'll mean making a copy for each thread/job.